src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -81,6 +84,40 @@ HDF5IOHandlerImpl::HDF5IOHandlerImpl(AbstractIOHandler* handler)
     H5Tinsert(m_H5T_CDOUBLE, "i", sizeof(double), H5T_NATIVE_DOUBLE);
     H5Tinsert(m_H5T_CLONG_DOUBLE, "r", 0, H5T_NATIVE_LDOUBLE);
     H5Tinsert(m_H5T_CLONG_DOUBLE, "i", sizeof(long double), H5T_NATIVE_LDOUBLE);
+
+    m_chunks = auxiliary::getEnvString( "OPENPMD_HDF5_CHUNKS", "auto" );
+    // JSON option can overwrite env option:
+    if( config.contains( "hdf5" ) )
+    {
+        m_config = std::move( config[ "hdf5" ] );
+
+        // check for global dataset configs
+        if( m_config.json().contains( "dataset" ) )
+        {
+            auto datasetConfig = m_config[ "dataset" ];
+            if( datasetConfig.json().contains( "chunks" ) )
+            {
+                m_chunks = datasetConfig.json()[ "chunks" ];
+            }
+            datasetConfig.declareFullyRead();
```

Comment:
```suggestion
```

`TracingJSON` does not trace beyond JSON arrays since those are a bit tricky to handle and it didn't matter so far. The purpose of `declareFullyRead` (and I forgot to document that) is to declare an array as fully read. By declaring the `datasetConfig` object as fully read, you will ignore wrong keys that the user passed, so you'll want to drop that.

Reply:
Oh, ok! Can you add these doc details in a follow-up? :)